template.js

```diff
@@ -9,71 +9,81 @@ const JSON = require('JSON');
 const logToConsole = require('logToConsole');
 const returnResponse = require('returnResponse');
 const setCookie = require('setCookie');
-const templateDataStorage = require('templateDataStorage');
 const sendHttpGet = require('sendHttpGet');
-const getTimestampMillis = require('getTimestampMillis');
 const sendHttpRequest = require('sendHttpRequest');
 const setResponseBody = require('setResponseBody');
 const setResponseHeader = require('setResponseHeader');
 const setResponseStatus = require('setResponseStatus');
 const getContainerVersion = require('getContainerVersion');
 const getRemoteAddress = require('getRemoteAddress');
+const getType = require('getType');
 const path = getRequestPath();
 
-// Check if this Client should serve exponea.js file
-if (path === data.proxyJsFilePath) {
-    claimRequest();
+const cookieWhiteList = ['xnpe_' + data.projectToken, '__exponea_etc__', '__exponea_time2__'];
+const headerWhiteList = ['referer', 'user-agent', 'etag'];
 
-    const now = getTimestampMillis();
-    const thirty_minutes_ago = now - (30 * 60 * 1000);
 
-    if (templateDataStorage.getItemCopy('exponea_js') == null || templateDataStorage.getItemCopy('exponea_stored_at') < thirty_minutes_ago) {
-        sendHttpGet('https://api.exponea.com/js/exponea.min.js', {headers: {'X-Forwarded-For': getRemoteAddress()}}).then((result) => {
-            if (result.statusCode === 200) {
-                templateDataStorage.setItemCopy('exponea_js', result.body);
-                templateDataStorage.setItemCopy('exponea_headers', result.headers);
-                templateDataStorage.setItemCopy('exponea_stored_at', now);
-            }
-            sendProxyResponse(result.body, result.headers, result.statusCode);
-        });
-    } else {
-        sendProxyResponse(
-            templateDataStorage.getItemCopy('exponea_js'),
-            templateDataStorage.getItemCopy('exponea_headers'),
-            200
-        );
-    }
+// Check if this Client should serve exponea.js file
+if (splitAndSanitize(data.proxyJsFilePath).reduce((res,proxyPath)=>res || equalOrStartsOrEnds(path,proxyPath), false)) {
+    claimRequest();
+    log({
```

Comment:
Please, use the Stape logging standard. It is described here https://github.com/snowplow/snowplow-gtm-server-side-tag/issues/6

Soon there will be official repo with a log standard description.